applications/StructuralMechanicsApplication/structural_mechanics_application.cpp

```diff
@@ -52,6 +52,9 @@ KratosStructuralMechanicsApplication::KratosStructuralMechanicsApplication():
     mShellThinCorotationalElement3D3N( 0, Element::GeometryType::Pointer( new Triangle3D3 <Node<3> >( Element::GeometryType::PointsArrayType( 3 ) ) ), true ),
     // Adding the membrane element
     mMembraneElement3D3N( 0, Element::GeometryType::Pointer( new Triangle3D3 <Node<3> >( Element::GeometryType::PointsArrayType( 3 ) ) ) ),
+	mMembraneElement3D4N( 0, Element::GeometryType::Pointer( new Quadrilateral3D4 <Node<3> >( Element::GeometryType::PointsArrayType( 4 ) ) ), false ),	
```

Comment:
IMPORTANT ONE: I believe that the existing membrane element implementation was already generic in the implementation, so that it could employ different type of geoemtries. Why didn't u base your implementation on that?
in which sense the new implementation is superior?